Add GCP documentation (#127)
DefangLabs/defang-docs: docs/tutorials/deploy-to-gcp.mdx

# Deploying to Google Cloud Platform (GCP) With Defang

> This tutorial will show you how to deploy your services to your own AWS account using Defang.

Reviewer: AWS should be GCP?

> This tutorial will show you how to deploy your services to your own AWS account using Defang.

Author: ```suggestion
This tutorial will show you how to deploy your services to your own GCP project using Defang.
```

## Pre-requisites
* [A Dockerfile in your project](https://docs.docker.com/get-started/docker-concepts/building-images/writing-a-dockerfile/)
* [A `compose.yaml` file in your project](https://docs.docker.com/compose/gettingstarted/)
* [A Defang Account](/docs/concepts/authentication)
* [The Defang CLI](/docs/getting-started#install-the-defang-cli)
* [GCP Account Credentials](https://cloud.google.com/docs/authentication)

## Step 1 - Navigate to your project directory
Head to the folder where your project is located.
```bash
$ cd path/to/your/project
```

## Step 2 - Authenticate Defang to use your GCP Account

> After signing up for your GCP account, select an existing project or [create a new project](https://developers.google.com/workspace/guides/create-project). Make sure [billing is enabled](https://cloud.google.com/billing/docs/how-to/modify-project). Then, note down the project ID and set it as environment variable `GCP_PROJECT_ID`.

Reviewer: Signing up or signing in?

> After signing up for your GCP account, select an existing project or [create a new project](https://developers.google.com/workspace/guides/create-project). Make sure [billing is enabled](https://cloud.google.com/billing/docs/how-to/modify-project). Then, note down the project ID and set it as environment variable `GCP_PROJECT_ID`.

Author: ```suggestion
After signing in for your GCP account, select an existing project or [create a new project](https://developers.google.com/workspace/guides/create-project). Make sure [billing is enabled](https://cloud.google.com/billing/docs/how-to/modify-project). Then, note down the project ID and set it as environment variable `GCP_PROJECT_ID`.
```

> After signing in for your GCP account, select an existing project or [create a new project](https://developers.google.com/workspace/guides/create-project). Make sure [billing is enabled](https://cloud.google.com/billing/docs/how-to/modify-project). Then, note down the project ID and set it as environment variable `GCP_PROJECT_ID`.

Reviewer: signing in *to* (not for)

Author: done

## Step 3 - Deploy

Invoke the `defang up` CLI command with the `--provider=gcp` flag or set the `DEFANG_PROVIDER=gcp` environment variable.

For example:

```bash
$ defang up --provider=gcp
```

## Step 4 - Inspect your deployment

> Defang will provision resources in your GCP account and deploy your services. You can inspect the resources created in your GCP dashboard.

Reviewer: Capitalize "Dashboard" and provide a link?

> Defang will provision resources in your GCP account and deploy your services. You can inspect the resources created in your GCP dashboard.

Author: ```suggestion
Defang will provision resources in your GCP account and deploy your services. You can inspect the resources created in your [GCP Dashboard](https://console.cloud.google.com/).
```
